src/cascadia/TerminalSettingsEditor/Actions.cpp

```diff
@@ -84,7 +87,10 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
 
     void KeyBindingViewModel::AttemptAcceptChanges(hstring newKeyChordText)
     {
-        auto args{ make_self<ModifyKeyBindingEventArgs>(_Keys, _Keys, _CurrentAction, unbox_value<hstring>(_ProposedAction)) };
+        // NewlyAdded: introduce a new key binding
+        // Otherwise:  include the old and new key binding data

```

Comment:
Mildly confused on "include the old and new key binding data."

Do you mean... create a fresh action... but ensure both key chords are a part of it? And it's fine if there are >2?

And how does passing `_Keys` twice accomplish that? Is it writing back out/updating the first param?

Reply:
So, we're overloading the responsibility of this function a bit. This is called when the ✅ button is clicked.
- **General Case**:
   - event args hold "old" keys, "new" keys, "old" action, and "new" action.
   - the event handler in `Actions` updates the settings model
- **When it's a new action (from the "add new" button)**: 
   - `_IsNewlyAdded` is `true`
   - there is no such thing as "old" keys and "old" action, so we just pass in "new" keys and "new" action.
   - the event handler in `Actions` updates the settings model in its own way

Reply:
Okay... so now my understanding is that it's "an array of keys before" and "an array of keys after"... Not 1 and 1 making >2 a potential issue.

But I still am missing why passing `_Keys` for before and after is OK. Shouldn't it have the same value? So nothing changes?

Reply:
> But I still am missing why passing `_Keys` for before and after is OK. Shouldn't it have the same value? So nothing changes?

Ah, because `_Keys` gets set in the next line:

https://github.com/microsoft/terminal/blob/b1209ae7a532a066f89ed35c806ad134512d560c/src/cascadia/TerminalSettingsEditor/Actions.cpp#L96-L112

Reply:
This code snippet doesn't refer to `_Keys`... is the magic behind the `NewKeys` method?

Reply:
Discussed offline. Made that section more clear.